app/include/zmk/led_indicators.h

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright (c) 2022 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#pragma once
+
+#include <zmk/endpoints_types.h>
+#include <zmk/hid.h>
+#include <zmk/led_indicators_types.h>
+
+zmk_leds_flags_t zmk_leds_get_current_flags();
+zmk_leds_flags_t zmk_leds_get_flags(enum zmk_endpoint endpoint, uint8_t profile);
+void zmk_leds_update_flags(zmk_leds_flags_t leds, enum zmk_endpoint endpoint, uint8_t profile);
+
+void zmk_leds_process_report(struct zmk_hid_led_report_body *report, enum zmk_endpoint endpoint,
+                             uint8_t profile);
```

Comment:
@joelspadin, should this be looked at as something that ZMK could/should provide independently of this PR?

The general idea is that some behaviors/functionalities may want to store data separately for each connected device: USB, BLE(0), BLE(1), ... so having a function to index the endpoints+profiles would be convenient.
For example I ran into this when looking into saving (and automatically restoring) base layers per device.

Basically we'd want something like:
- `zmk_endpoint_index_t zmk_endpoint_index(zmk_endpoint_select)`
- `zmk_endpoint_select zmk_current_endpoint()`
- maybe `zmk_endpoint_index_t zmk_current_endpoint_index()` to shortcut `zmk_endpoint_index(zmk_current_endpoint())`
- `ZMK_ENDPOINT_COUNT` (`ZMK_USB_PROFILE_COUNT + ZMK_BLE_PROFILE_COUNT` + possibly future endpoint count)

I don't know enough to tell what the right way to do it is, just pointing out that it's worth considering the wider pattern/use-case.

Reply:
@petejohanson Any thoughts on how to best design this?

Reply:
So, I generally really like the idea of a tagged union for this, in terms of the data type for identifying a given endpoint (active or otherwise). 

As I think on it, I think there's maybe use in having clear concept of "endpoint transport", e.g. USB, BLE, and then figuring out a way to have "per transport selections/preferences" that would allow us to capture BLE profile index, etc.

Doing it this way would allow us to have both a "preferred endpoint" and "active endpoint" nicely modeled, and also open up the way to have each transport offer an away to enumerate possible endpoints available on that transport.

And I do agree, this is something that should be really nailed down first in a separate PR, for focus and clarity while we independently review and work out issues w/ LED indicators.

Reply:
For the terminology, what do you think of "endpoint transport" (USB/BLE) and "endpoint instance" (USB, BLE profile 0, etc.)?

I could put together a PR for that if you'd like.

Reply:
Seems good to me.

Reply:
@bortoz Try this and let me know if it works for you: https://github.com/zmkfirmware/zmk/pull/1803

Reply:
Thank you! I've rebased on your branch.